src/lib/content.d.ts

```diff
@@ -443,3 +445,103 @@ interface CampaignFieldSelect {
     hideLabel: boolean;
     label: string;
 }
+
+// -------------------------------------
+// preNormalisedCampaign callout types
+// -------------------------------------
+
+type preNormalizedCampaignType = {
+    id: string;
+    name: string;
+    rules: [];
+    priority: number;
+    activeFrom: number;
+    displayOnSensitive: boolean;
+    fields: {
+        formId: number;
+        callout: string;
+        _type: string;
+        description: string;
+        tagName: string;
+        formFields: preNormalizedCampaignFieldType[];
+    };
+};
+
+type preNormalizedCampaignFieldType =
+    | preNormalizedCampaignFieldText
+    | preNormalizedCampaignFieldTextArea
+    | preNormalizedCampaignFieldFile
+    | preNormalizedCampaignFieldRadio
+    | preNormalizedCampaignFieldCheckbox
+    | preNormalizedCampaignFieldSelect;
+
+type preNormalizedCampaignFieldText = {
```

Comment:
You can create an `formInput` interface and `extends` it for each input for the common properties

Reply:
good point! I think I will break this out in another PR bec we have quite a bit of type logic changed here